app.py

```diff
@@ -1,24 +1,115 @@
-from flask import url_for,render_template,redirect,Flask,flash
-from forms import LoginForm
+from flask import Flask, render_template, flash, redirect, url_for, session, request, logging
+from passlib.hash import sha256_crypt
+from functools import wraps
+from forms.py import RegisterForm 
 from Make_Tables.mysqlconnect import mydb, mycursor, create_insert_statement   #Imported the mysqlconnect.py file from Make_tables folder
 
+
 app=Flask(__name__,static_url_path='/public')
 app.config['SECRET_KEY']='c828b6ff21f45063fd7860e5c1b1d233'
 
-@app.route('/')
-def home():
-    return render_template('index.html')
 
-@app.route("/login",methods=['GET','POST'])
-def Login():
-    form=LoginForm()
-    if form.validate_on_submit():
+
+# User Registration
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    form = RegisterForm(request.form)
+    if request.method == 'POST' and form.validate():
+        
+        userID = form.userID.data
         if form.email.data[1:6].isdecimal() and form.email.data[6:]=='@students.iitmandi.ac.in' :
-            flash('You have logged-in successfully!')
-            return redirect(url_for('home'))
+            email = form.email.data
+        name = form.name.data
+        password = sha256_crypt.encrypt(str(form.password.data))
+        admin boolean = 0
+        
+
+        # Execute query
+        mycursor.execute("INSERT INTO users(userID, email, password, name, admin boolean) VALUES(%s, %s, %s, %s, %s)", (userID, email, password, name, admin boolean))
+
+        # Commit to DB
+        mydb.commit()
+
+        
+        flash('You are now registered and can log in!', 'success')
+
+        return redirect(url_for('login'))
+    return render_template('register.html', title='Register | SAC Portal, IIT Mandi', form=form)
+
+
+
+# User login
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    if request.method == 'POST':
+
+        # Get Form Fields
+        userID = request.form['userID']
+        password_candidate = request.form['password']
+
+        
+
+        # Get user by userID
+        result = mycursor.execute("SELECT * FROM users WHERE userID = %s", [userID])
+
+
+        if result > 0:
+            # Get stored hash
+            data = mycursor.fetchone()
+            password = data['password']
+
+            # Compare Passwords
+            if sha256_crypt.verify(password_candidate, password):
+                # Passed
+                session['logged_in'] = True
+                session['userID'] = userID
+                session['email'] = email
+                session['admin status'] = admin boolean
+
+                flash('You are now logged in!', 'success')
+                return redirect(url_for('dashboard'))
+            else:
+                error = 'Invalid login!Please enter valid password!'
+                return render_template('login.html', error=error)
```

Comment:
- [ ] don't use render_template for now, since we have decided on doing backend and frontend independently, the absence of these files make it difficult to test